uefi-raw/src/protocol/shell.rs

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! EFI Shell Protocol v2.2
+
+use core::ffi::c_void;
+
+use crate::{Char8, Char16, Event, Guid, Handle, Status, guid};
+
+use super::device_path::DevicePathProtocol;
+use super::file_system::FileInfo;
+use super::shell_params::ShellFileHandle;
+
+/// List Entry for File Lists
+#[derive(Debug)]
+#[repr(C)]
+pub struct ListEntry<'a> {
+    pub f_link: *mut ListEntry<'a>,
+    pub b_link: *mut ListEntry<'a>,
+}
+
+/// ShellFileInfo for File Lists
+#[derive(Debug)]
+#[repr(C)]
+pub struct ShellFileInfo<'a> {
+    pub link: ListEntry<'a>,
+    pub status: Status,
+    pub full_name: *mut Char16,
+    pub file_name: *mut Char16,
+    pub shell_file_handle: Handle,
+    pub file_info: FileInfo,
+}
+
+/// Used to specify where component names should be taken from
+pub type ShellDeviceNameFlags = u32;
+pub const DEVICE_NAME_USE_COMPONENT_NAME: u32 = 0x0000001;
+pub const DEVICE_NAME_USE_DEVICE_PATH: u32 = 0x0000002;
```

Comment:
Use the `bitflags` macro for this type and constants. See for example https://github.com/rust-osdev/uefi-rs/blob/main/uefi-raw/src/protocol/console/serial.rs#L6.